proposed/http-factory/http-factory.md

```diff
@@ -77,24 +82,30 @@ Has the ability to create server requests.
 ```php
 namespace Psr\Http\Message;
 
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Message\UriInterface;
+
 interface ServerRequestFactoryInterface
 {
     /**
      * Create a new server request.
      *
-     * @param string $method
-     * @param UriInterface|string $uri
+     * The method and URI of the request SHOULD be derived from the given
+     * server parameters.
      *
-     * @return ServerRequestInterface
-     */
-    public function createServerRequest($method, $uri);
-
-    /**
-     * Create a new server request from PHP globals.
+     * If the method or URI parameter is passed, these values MUST be used
```

Comment:
What should happen in this case:
```php
$factory->createServerRequest($server, null, $uri)
```

 I think that the intention is to use `$server` value instead of `null`. Nevertheless `these values MUST be used` is suggesting to me that the `null` must be used, therefore `\InvalidArgumentException` should be raised.